src/editor/keystroke_log.rs

```diff
@@ -71,27 +78,10 @@ pub struct KeyStrokeLog {
     /// The maximum number of entries that should be displayed at once
     max_entries: usize,
     /// The keystrokes that will be included in the next log entry
-    unlogged_keystrokes: Vec<Key>,
+    unlogged_keystrokes: Vec<KeyEvent>,
 }
-
-impl KeyStrokeLog {
-    /// Create a new (empty) keystroke log
-    pub fn new(max_entries: usize) -> KeyStrokeLog {
-        KeyStrokeLog {
-            keystrokes: vec![],
-            max_entries,
-            unlogged_keystrokes: vec![],
-        }
-    }
-
-    /// Sets and enforces the max entry limit
-    pub fn set_max_entries(&mut self, max_entries: usize) {
-        self.max_entries = max_entries;
-        self.enforce_entry_limit();
-    }
-
-    /// Draw a log of recent keystrokes to a given terminal at a given location
-    pub fn render(&self, term: &Term, row: usize, col: usize) {
+impl tui::widgets::Widget for &'_ KeyStrokeLog {
```

Comment:
Tiny nitpick: a blank line between the struct and `impl` would be nice - I'm not sure why rustfmt doesn't enforce this by default, but hey.